Godot is not compatible with OpenGL 2.1, but it should be compatible

**Operating system or device - Godot version:**

Godot 2.1 official / custom build from GIT; Windows 7; old Intel graphics which supports OpenGL 2.1:
```
Mobile Intel(R) 4 Series Express Chipset Family from Intel
OpenGL version 2.1.0 - Build 8.15.10.2869 is supported
```

It doesn't work also on VirtualBox drivers.

**Issue description** (what happened, and what was expected):

- Current behavior: Godot crashes at startup without any useful information.

- Expected behavior: Godot runs on this platform, because it can run there. Otherwise show information about incompatible OpenGL driver and close without crash.

**Steps to reproduce:**

Run Godot and expect crash on OpenGL 2.1 on Windows.

**Additional information:**

On Linux and Mesa OpenGL 2.1 drivers - Godot works properly (on the same hardware), because Mesa is new and Mesa is cheating :)

As far I found 11 function which are compatible with GLES 2.0 and OpenGL 3.0 (not OpenGL 2.1):
- `glGenFramebuffers`,
- `glGenerateMipmap`,
- `glGenRenderbuffers`,
- `glBindRenderbuffer`,
- `glBindFramebuffer`,
- `glCheckFramebufferStatus`,
- `glDeleteRenderbuffers `,
- `glDeleteFramebuffers`,
- `glFramebufferTexture2D`,
- `glRenderbufferStorage`,
- `glFramebufferRenderbuffer`.

Every function listed above are supported with "EXT" suffix (e.g. `glGenerateMipmapEXT`). Replacing it in `rasterizer_gles2.cpp` allows Godot to run and play/edit 2D game on old Intel driver (VirtualBox shows white window and GLSL link error in console, but doesn't crash).

Godot should use functions with `EXT` suffix for OpenGL < 3.0 and it should check every pointer on initialization (instead of checking the OpenGL version).

Works properly :)

But I've got crash on Intel on `3d/platformer` demo (`2d/platformer` works). Investigating (not my notebook, so it can take some time :) ). Crashes in ` RasterizerGLES2::_process_hdr()`.

About VirtualBox: this might be only VirtualBox GLSL bug, but I've got only white screen and GLSL link error:
```
ERROR: ShaderGLES2::get_current_version: CanvasShaderGLES2: Program LINK FAILED:
Vertex info
-----------
(94) : error C5145: must write to gl_Position
```

--- a/drivers/gles2/rasterizer_gles2.cpp
+++ b/drivers/gles2/rasterizer_gles2.cpp
@@ -10809,8 +10809,46 @@ void RasterizerGLES2::init() {
 		print_line(String("GLES2: Using GLEW ") + (const char*) glewGetString(GLEW_VERSION));
 	}
 
+	// Godot makes use of functions from ARB_framebuffer_object extension which is not implemented by all drivers.
+	// On the other hand, these drivers might implement the older EXT_framebuffer_object extension
+	// with which current source code is backward compatible.
+
+	bool framebuffer_object_is_supported = glewIsSupported("GL_ARB_framebuffer_object");
+
+	if ( !framebuffer_object_is_supported ) {
+		WARN_PRINT("GL_ARB_framebuffer_object not supported by your graphics card.");
+
+		if ( glewIsSupported("GL_EXT_framebuffer_object") ) {
+			// falling-back to the older EXT function if present
+			WARN_PRINT("Falling-back to GL_EXT_framebuffer_object.");
+
+			glIsRenderbuffer = glIsRenderbufferEXT;
+			glBindRenderbuffer = glBindRenderbufferEXT;
+			glDeleteRenderbuffers = glDeleteRenderbuffersEXT;
+			glGenRenderbuffers = glGenRenderbuffersEXT;
+			glRenderbufferStorage = glRenderbufferStorageEXT;
+			glGetRenderbufferParameteriv = glGetRenderbufferParameterivEXT;
+			glIsFramebuffer = glIsFramebufferEXT;
+			glBindFramebuffer = glBindFramebufferEXT;
+			glDeleteFramebuffers = glDeleteFramebuffersEXT;
+			glGenFramebuffers = glGenFramebuffersEXT;
+			glCheckFramebufferStatus = glCheckFramebufferStatusEXT;
+			glFramebufferTexture1D = glFramebufferTexture1DEXT;
+			glFramebufferTexture2D = glFramebufferTexture2DEXT;
+			glFramebufferTexture3D = glFramebufferTexture3DEXT;
+			glFramebufferRenderbuffer = glFramebufferRenderbufferEXT;
+			glGetFramebufferAttachmentParameteriv = glGetFramebufferAttachmentParameterivEXT;
+			glGenerateMipmap = glGenerateMipmapEXT;
+
+			framebuffer_object_is_supported = true;
+		}
+		else {
+			ERR_PRINT("Framebuffer Object is not supported by your graphics card.");
+		}
+	}
+
 	// Check for GL 2.1 compatibility, if not bail out
-	if (!glewIsSupported("GL_VERSION_2_1")) {
+	if (!(glewIsSupported("GL_VERSION_2_1") && framebuffer_object_is_supported)) {
 		ERR_PRINT("Your system's graphic drivers seem not to support OpenGL 2.1 / GLES 2.0, sorry :(\n"
 			  "Try a drivers update, buy a new GPU or try software rendering on Linux; Godot will now crash with a segmentation fault.");
 		OS::get_singleton()->alert("Your system's graphic drivers seem not to support OpenGL 2.1 / GLES 2.0, sorry :(\n"

